Tests/CredentialsJWTTests/TestRawRouteJWT.swift

```diff
@@ -0,0 +1,289 @@
+                }
+            })
+        }
+
+        // User 2 JWT created.  Users 2 and 3 created for later tests where multiple tokens are needed.
+        performServerTest(router: router) { expectation in
+            self.performRequest(method: "post", path: "/generaterawjwt", contentType: "application/json", callback: { response in
+                do {
+                    guard let body = try response?.readString() else {
+                        XCTFail("Couldn't read response")
+                        return
+                    }
+                    do {
+                        self.jwtString2 = body
+                    }
+                } catch {
+
+                }
+                expectation.fulfill()
+            }, requestModifier: { request in
+                do {
+                    try request.write(from: JSONEncoder().encode(self.testUser2))
+                } catch {
+                    XCTFail("Failed to send data")
+                }
+            })
+        }
+    }
+
+    // Tests that the pre-constructed JWT type maps correctly to the JWT decoded from
+    // the jwtString earlier defined.
+    func testDefaultTokenProfile() {
+        do {
+            let profileInstance = try JWT<TestClaims>(jwtString: jwtString, verifier: .hs256(key: key!))
+            // An equivalent test profile, constructed directly.
+            let testTokenProfile = JWT(claims: TestClaims(sub: "Test"))
+            XCTAssertEqual(profileInstance.claims, testTokenProfile.claims, "The reference JWT instance did not match the instance decoded from the jwt string")
+        } catch {
+            XCTFail("error")
+        }
+    }
+
+    // Tests that when a correct token is supplied a valid UserProfile is created.
+    func testCorrectToken() {
+        performServerTest(router: router) { expectation in
+            self.performRequest(method: "get", path: "/rawtokenauth", callback: { response in
+                XCTAssertNotNil(response, "ERROR!!! ClientRequest response object was nil")
+                XCTAssertEqual(response?.statusCode, HTTPStatusCode.OK, "HTTP Status code was \(String(describing: response?.statusCode))")
+                do {
+                    guard let body = try response?.readString() else {
+                        XCTFail("No response body")
+                        return
+                    }
+                    let profile = body
+                    let testProfile = UserProfile(id: "Test", displayName: "Test", provider: "JWT")
+                    XCTAssertEqual(profile, testProfile.id)
+                } catch {
+                    XCTFail("Could not decode response: \(error)")
+                }
+                expectation.fulfill()
+            }, headers: ["X-Token-Type" : "JWT", "Authorization" : "Bearer " + self.jwtString])
+        }
+    }
+
+    // Tests that when an incorrect token is supplied a valid token but wrong UserProfile is created.
+    func testIncorrectToken() {
```

Comment:
This test appears to just test that a second (valid) JWT results in a different ID being returned compared to the `testCorrectToken` above.  I suppose there's value in that it proves that the response on this route is determined by the JWT claims (not a static response).  How about:
- rename the test to `testCorrectToken2()`
- assert that the profile contains "Test2"